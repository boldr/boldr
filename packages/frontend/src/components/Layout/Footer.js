// @flow
import React from 'react';
import cn from 'classnames';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  className?: string,
};

export function Footer({ tag = 'footer', ...props }: Props) {
  const className = cn('boldr-footer', props.className);

  return React.createElement(tag, { ...props, className });
}

export default createWrappedComponent(Footer);
