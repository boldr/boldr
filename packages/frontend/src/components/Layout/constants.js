export const BREAKPOINT_NAMES = ['xs', 'sm', 'md', 'lg'];

export const THEME_NAME = 'boldrgrid';

export const GRID_SETTINGS = {
  gridSize: 12,
  gutterWidth: 1,
  outerMargin: 2,
  container: {
    sm: 46,
    md: 61,
    lg: 76,
  },
  breakpoints: {
    xs: 0,
    sm: 48,
    md: 64,
    lg: 75,
  },
};
