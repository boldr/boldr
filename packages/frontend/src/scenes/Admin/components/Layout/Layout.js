/* @flow */
import * as React from 'react';
import { Hero, HeroHeader, HeroBody, HeroFooter } from '@boldr/ui/Hero';
// internal
import Navigation from '../Navigation';
import DashboardMain from '../DashboardMain';
import DashboardFooter from '../DashboardFooter';

export type Props = {
  children: React.ChildrenArray<number>,
  location: Object,
};

function Layout(props: Props) {
  return (
    <Hero isColor="info" className="boldrui-admin-layout" isFullHeight>
      <HeroHeader>
        <Navigation location={props.location} />
      </HeroHeader>
      <HeroBody>
        <DashboardMain>
          {props.children}
        </DashboardMain>
      </HeroBody>
      <HeroFooter>
        <DashboardFooter />
      </HeroFooter>
    </Hero>
  );
}
export default Layout;
