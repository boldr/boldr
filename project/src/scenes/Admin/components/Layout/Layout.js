/* @flow */
import React from 'react';
import type { Node } from 'react';
import { Hero, HeroHeader, HeroBody, HeroFooter } from '@boldr/ui/Hero';
// internal
import type { RouterLocation } from '../../../../types/boldr';
import Navigation from '../Navigation';
import DashboardMain from '../DashboardMain';
import DashboardFooter from '../DashboardFooter';

export type Props = {
  children: Node,
  location: RouterLocation,
};

function Layout(props: Props) {
  return (
    <Hero isColor="info" className="boldr-admin-layout" isFullHeight>
      <HeroHeader>
        <Navigation location={props.location} />
      </HeroHeader>
      <HeroBody>
        <DashboardMain>{props.children}</DashboardMain>
      </HeroBody>
      <HeroFooter>
        <DashboardFooter />
      </HeroFooter>
    </Hero>
  );
}
export default Layout;
