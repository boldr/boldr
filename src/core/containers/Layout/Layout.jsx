import React from 'react';

export const LayoutContainer = ({ children }) => (
  <div>
      {children}
  </div>
);

LayoutContainer.propTypes = {
  children: React.PropTypes.element.isRequired
};

export default LayoutContainer;
