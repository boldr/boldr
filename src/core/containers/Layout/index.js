import LayoutContainer from 'core/containers/Layout/Layout.jsx';
export default LayoutContainer;
