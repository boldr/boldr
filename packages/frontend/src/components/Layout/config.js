import { css } from 'styled-components';
import { GRID_SETTINGS, THEME_NAME } from './constants';

function insertMediaQuery(media) {
  return (...args) => css`
    @media ${media} {
      ${css(...args)}
    }
  `;
}

const configCache = [];
const makeCacheId = props => JSON.stringify((props.theme && props.theme[THEME_NAME]) || {});
const resolveConfig = props => {
  const themeConf = (props.theme && props.theme[THEME_NAME]) || {};

  const conf = {
    ...GRID_SETTINGS,
    ...themeConf,
    container: {
      ...GRID_SETTINGS.container,
      ...themeConf.container,
    },
    breakpoints: {
      ...GRID_SETTINGS.breakpoints,
      ...themeConf.breakpoints,
    },
  };

  conf.media = Object.keys(conf.breakpoints).reduce((media, breakpoint) => {
    const breakpointWidth = conf.breakpoints[breakpoint];
    media[breakpoint] = insertMediaQuery(
      'only screen' + (breakpoint === 0 ? '' : ` and (min-width: ${breakpointWidth}em)`),
    );
    return media;
  }, {});

  return conf;
};

export default function config(props) {
  const cacheId = makeCacheId(props);
  if (configCache[0] === cacheId) {
    return configCache[1];
  }

  const conf = resolveConfig(props);

  configCache[0] = cacheId;
  configCache[1] = conf;

  return conf;
}
