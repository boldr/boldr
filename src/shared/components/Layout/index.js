import Grid from './Grid'
import Row from './Row'
import Col from './Col'
import { BASE_CONF } from './config'

export {
  Grid,
  Row,
  Col,
  BASE_CONF
}
