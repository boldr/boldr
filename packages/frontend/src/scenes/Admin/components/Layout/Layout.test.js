import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';

import Layout from './Layout';

describe('Layout', () => {
  it('renders snapshot', () => {
    const container = shallow(<Layout>Layout</Layout>);
    expect(shallowToJson(container)).toMatchSnapshot();
  });
});
