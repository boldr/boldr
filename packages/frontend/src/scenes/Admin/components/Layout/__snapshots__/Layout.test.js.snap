// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders snapshot 1`] = `
<boldruiComponent(Hero)
  className="boldrui-admin-layout"
  isColor="info"
  isFullHeight={true}
>
  <boldruiComponent(HeroHeader)>
    <Navigation />
  </boldruiComponent(HeroHeader)>
  <boldruiComponent(HeroBody)>
    <DashboardMain__DashboardMain>
      Layout
    </DashboardMain__DashboardMain>
  </boldruiComponent(HeroBody)>
  <boldruiComponent(HeroFooter)>
    <DashboardFooter />
  </boldruiComponent(HeroFooter)>
</boldruiComponent(Hero)>
`;
